gym_electric_motor/physical_systems/mechanical_loads/polynomial_static_load.py

```diff
@@ -47,18 +53,26 @@ def __init__(self, load_parameter=None, limits=None, load_initializer=None):
         self._c = self._load_parameter['c']
 
     def _static_load(self, omega, *_):
-        """
-        Calculation of the load torque for a given speed omega.
-        """
-        sign = 1 if omega > 0 else -1 if omega < 0 else 0
-        return sign * (self._c * omega**2 + self._b * abs(omega) + self._a)
+        """Calculation of the load torque for a given speed omega."""
+        sign = 1 if omega > 0 else -1 if omega < -0 else 0
+        # Limit the constant load term 'a' for velocities around zero for a more stable integration
+        a = sign * self._a \
+            if abs(omega) > self._a / self._j_total * self.tau_decay \
+            else self._j_total / self.tau_decay * omega
+        return sign * self._c * omega**2 + self._b * omega + a
 
     def mechanical_ode(self, t, mechanical_state, torque):
         # Docstring of superclass
-        return np.array([(torque - self._static_load(mechanical_state[self.OMEGA_IDX])) / self._j_total])
+        omega = mechanical_state[self.OMEGA_IDX]
+        static_load = self._static_load(omega)
+        load = torque - static_load
+        return np.array([load / self._j_total])
 
     def mechanical_jacobian(self, t, mechanical_state, torque):
         # Docstring of superclass
-        sign = 1 if mechanical_state[self.OMEGA_IDX] > 0 else -1 if mechanical_state[self.OMEGA_IDX] < 0 else 0
-        return np.array([[(-self._b * sign - 2 * self._c * mechanical_state[self.OMEGA_IDX])/self._j_total]]), \
-            np.array([1/self._j_total])
+        omega = mechanical_state[self.OMEGA_IDX]
+        sign = 1 if omega > 0 else -1 if omega < 0 else 0
+        # Linear region of the constant load term 'a' ?
+        a = 0 if abs(omega) > self._a * self.tau_decay / self._j_total else self._j_total / self.tau_decay
+        return np.array([[(-self._b * sign - 2 * self._c * omega + a) / self._j_total]]), \
```

Comment:
Shouldn't "a" have a negative sign as well?

Reply:
No, the derivative of the load ODE over omega is calculated at those lines. In the old version, the constant torque "a" has been completely eliminated by the derivation (except for a non-modeled infinite slope at omega=0). 
Now, there is a non-infinite slope between the two levels of "a" for negative and positive speeds. 
Therefore, the derivative of 'a' at that point is constant `self._j_total / self._tau_decay (>0)`.